_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,344 @@
+categories: news 
+authors:
+- joemoe:
+  name: "Johannes Moser"
+
+excerpt: The Apache Flink community is excited to announce the release of Flink 1.14.0! More than 200 contributor worked on over 1,000 issues. The release brings exciting new features like a more seamless streaming/batch integration, automatic network memory tuning, a hybrid source to switch data streams between storgage systems (e.g., Kafka/S3), Fine-grained resource management, PyFlink performance and debugging enhancements, and a Pulsar connector. 
+
+---
+
+The Apache Software Foundation recently released its annual report and Apache Flink once again made
+it on the list of the top 5 most active projects! This remarkable
+activity also shows in the new 1.14.0 release. Once again, more than 200 contributors worked on
+over 1,000 issues. We are proud of how this community is consistently moving the project forward.
+
+This release brings many new features and improvements in areas such as the SQL API, more connector support, checkpointing, and PyFlink.
+A major area of changes in this release is the integrated streaming & batch experience. We believe
+that, in practice, unbounded stream processing goes hand-in-hand with bounded- and batch processing tasks in practice,
+because many use cases require processing historic data from various sources alongside streaming data.
+Examples are data exploration when developing new applications, bootstrapping state for new applications, training
+models to be applied in a streaming application, re-processing data after fixes/upgrades, and .
+
+In Flink 1.14, we finally made it possible to **mix bounded and unbounded streams in an application**:
+Flink now supports taking checkpoints of applications that are partially running and partially finished (some
+operators reached the end of the bounded inputs). Additionally, **bounded streams now take a final checkpoint**
+when reaching their end to ensure smooth committing of results in sinks.
+
+The **batch execution mode now supports programs that use a mixture of the DataStream API and the SQL/Table API**
+(previously only pure Table/SQL or DataStream programs).
+
+The unified Source and Sink APIs have gotten an update, and we started **consolidating the connector ecosystem around the unified APIs**. We added a new **hybrid source** can bridge between multiple storage systems.
+You can now do things like read old data from Amazon S3 and then switch over to Apache Kafka.
+
+Furthermore, this release takes another step in our initiative of making Flink more self-tuning and
+to require less Stream-Processor-specific knowledge to operate. We started that initiative in the previous
+release with [Reactive Scaling](/news/2021/05/03/release-1.13.0.html#reactive-scaling) and are now
+adding **automatic network memory tuning** (*a.k.a. Buffer Debloating*). This feature speeds up checkpoints
+under load without sacrificing performance or increasing checkpoint size, by continuously adjusting the
```

Comment:
> without sacrificing performance

That's unfortunately not true. It's causing a couple of % slow down, that might be hard to get rid of. What's worse, is that it looks like this performance drop is visible even in heavily back pressured jobs, with low throughputs and long time to process single record. In other words, it doesn't slow down max theoretical throughput, but potentially it might slow down every job.

Reply:
I'll make a small adjustment.